framework/web/UrlRule.php

```diff
@@ -352,4 +352,16 @@ protected function getParamRules()
     {
         return $this->_paramRules;
     }
+
+    /**
+     * Remove numeric indexes from the response
+     * @param $params
+     *
+     * @return mixed|string
+     */
+    protected function custom_http_build_query($params) {
```

Comment:
Avoid code duplicates. Make this public and static, possibly move to some helper class.